diff -ruN mrfix/dialogs/MREditProfilesPanel.cpp mrfix_mod/dialogs/MREditProfilesPanel.cpp
--- mrfix/dialogs/MREditProfilesPanel.cpp	2026-04-08 20:37:52.000000000 +0000
+++ mrfix_mod/dialogs/MREditProfilesPanel.cpp	2026-04-08 20:51:04.709809205 +0000
@@ -331,7 +331,7 @@
 	addPanelLabel(dialog, TRect(g.labelLeft + 1, g.formatLineY, g.inputLeft - 2, g.formatLineY + 1),
 	              "Format line:");
 	formatLineField_ = addPanelInput(dialog,
-	                               TRect(g.inputLeft, g.formatLineY, g.inputRight, g.formatLineY + 1),
+	                               TRect(g.inputLeft, g.formatLineY, browseFieldRight - 2, g.formatLineY + 1),
 	                               kFormatLineFieldSize - 1);
 
 	addPanelLabel(dialog, TRect(g.optionsHeadingX, g.optionsHeadingY, config_.dialogWidth - 2, g.optionsHeadingY + 1),
